### PR TITLE
[exporter/elasticsearch] Change default retry.retry_to_status to [429]

### LINKED ISSUE #32584: [exporter/elasticsearch] Make retry statuses configurable and default to 429 only
### Component(s)

exporter/elasticsearch

### Is your feature request related to a problem? Please describe.

Elasticsearch exporter retries on 500, 502, 503, 504, 429 by default, on both request level and document level. This is not ideal and may lead to duplicates.

### Describe the solution you'd like

The retriable statuses should be configurable such that users may decide whether they prefer guaranteed delivery or no duplicates. It should default to retrying on 429 only.

### Additional context

Integration tests need to be updated since it is testing with document level 500s now.

### PR DESCRIPTION
**Pull request by carsonip**
**Description:** <Describe what has changed.>

Breaking change to change default retry_on_status to 429

**Link to tracking Issue:** <Issue number if applicable>
Fixes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32584

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: .chloggen/elasticsearchexporter_retry-on-status-429.yaml
Comment (crobert-1) [inline .chloggen/elasticsearchexporter_retry-on-status-429.yaml:18]: It might be worthwhile to add a note here to users on how to preserve previous behavior.

**Review by crobert-1 — APPROVED** [review]

**Comment by crobert-1** [comment]
Can you also add to your PR description that this is related to (or even `Resolves` if true) https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32584?

File: .chloggen/elasticsearchexporter_retry-on-status-429.yaml
Comment (carsonip) [reply .chloggen/elasticsearchexporter_retry-on-status-429.yaml:18]: Updated, thanks!

**Comment by crobert-1** [comment]
My approval stands, if we could get another code owner approval that would be great. Otherwise I can add `ready to merge`  after a few more days have passed to give others a chance to review.

File: .chloggen/elasticsearchexporter_retry-on-status-429.yaml
Comment (andrzej-stencel) [inline .chloggen/elasticsearchexporter_retry-on-status-429.yaml:19]: This needs to be a valid Yaml, so either put the key and value on the same line, or use on of the [Yaml block style indicators](https://yaml.org/spec/1.2.2/#chapter-8-block-style-productions) like `|`.

```suggestion
subtext: To retain the previous behavior, set retry.retry_on_status to `[429, 500, 502, 503, 504]`.
```

**Review by andrzej-stencel — APPROVED** [review]

File: .chloggen/elasticsearchexporter_retry-on-status-429.yaml
Comment (andrzej-stencel) [reply .chloggen/elasticsearchexporter_retry-on-status-429.yaml:19]: Actually, looks like the changelog tool does not have any problem with that, so there's no need to fix this.

**Pull request merged**
